src/libra_py/auxiliary_functions.py

```diff
@@ -0,0 +1,547 @@
+#*********************************************************************************
```

Comment:
this file has the implementation of the very specific method, for your workflow, so it shall not be at this (more general) level, and shall be inside workflows/nbra